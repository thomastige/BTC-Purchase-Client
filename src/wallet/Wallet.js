import React from 'react';

import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

/*
Wallet component. Contains the current amounts available in the user's wallet in both USD and BTC.
*/
class Wallet extends React.Component {
	constructor(props) {
		super(props);
	}

   render() {
      return (
         <div>
			<p>Account Balance</p>
			<table id="wallet">
				<tbody>
					<tr>
						<td className="walletRowTitle">
							USD
						</td>
						<td>
							{this.props.usdValue}
						</td>
					</tr>
					<tr>
						<td className="walletRowTitle">
							BTC
						</td>
						<td>
							{this.props.btcValue}
						</td>
					</tr>
				</tbody>
			</table>
		 </div>
      );
   }
}
function mapStateToProps(state){
	return {
		usdValue: parseFloat(state.walletReducer.usd).toFixed(2),
		btcValue: parseFloat(state.walletReducer.btc).toFixed(8)
	};
}
export default connect(mapStateToProps)(Wallet);