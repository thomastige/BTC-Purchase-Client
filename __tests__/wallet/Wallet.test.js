import React from 'react';
import Wallet from 'Wallet';
import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store'

Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureStore();
let store;

test('Testing Wallet snapshot', () => {
	store = mockStore({walletReducer: {usd: 0, btc: 0}});
	const wrapper = shallow(<Wallet store={store}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Testing Wallet values', () => {
	var usdValue = '150.00';
	var btcValue = '12.00000000';
	store = mockStore({walletReducer: {usd: usdValue, btc: btcValue}});
	const wrapper = mount(<Wallet store={store}/>);
	expect(wrapper.find('td').at(1).html()).toBe('<td>' + usdValue + '</td>');
	expect(wrapper.find('td').at(3).html()).toBe('<td>' + btcValue + '</td>');
});