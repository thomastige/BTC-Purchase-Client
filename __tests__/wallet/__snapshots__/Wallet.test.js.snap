// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Wallet snapshot 1`] = `
<Wallet
  btcValue="0.00000000"
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  usdValue="0.00"
/>
`;
